### PR TITLE
fix/ GatewayHttpClient not resolving base_url for Gateway service

### PR DESCRIPTION
**Pull request by CrimsonJacket**
**Before submitting this PR, please make sure**:

- [x] Your code builds clean without any errors or warnings
- [x] You are using approved title ("feat/", "fix/", "docs/", "refactor/")

**A description of the changes proposed in the pull request**:
Refactor `GatewayHttpClient` as a Singleton class. Modify all locations where `gateway_http_client` is called.

[sc-24690]

**Tests performed by the developer**:

**Tips for QA testing**:

### CODE REVIEW

File: hummingbot/strategy/amm_arb/amm_arb.py [hunk 1]
```diff
@@ -187,6 +189,21 @@ def apply_budget_constraint(self, arb_proposals: List[ArbProposal]):
...
+        active_markets: List[str] = [side.market_info.market.name for side in proposal_sides]
+        if not all([market in AllConnectorSettings.get_gateway_evm_amm_connector_names() for market in active_markets]):
+            return arb_proposal
+
+        results = []
+        for idx, side in enumerate(proposal_sides):
+            pass
+
+        return results
+
```
Comment (vic-en) [inline hummingbot/strategy/amm_arb/amm_arb.py:206]: I guess this is still a WIP. Right?

**Review by vic-en — CHANGES_REQUESTED** [review]
I updated the unit test to use include `get_instance()`

File: hummingbot/strategy/amm_arb/amm_arb.py (hunk 1 above)
Comment (CrimsonJacket) [reply hummingbot/strategy/amm_arb/amm_arb.py:206]: Oh yes, I thought i removed it already. Will make the necessary changes

**Review by vic-en — APPROVED** [review]
All changes LGTM

**Pull request merged**